test/integration/study_creation_test.rb

```diff
@@ -108,23 +118,48 @@ class StudyCreationTest < ActionDispatch::IntegrationTest
     assert_equal 19, study.genes.size, 'Did not parse all genes from expression matrix'
 
     # verify that counts are correct, this will ensure that everything uploaded & parsed correctly
-    study.reload
     gene_count = study.gene_count
     cluster_count = study.cluster_groups.size
     metadata_count = study.cell_metadata.size
     cluster_annot_count = study.cluster_annotation_count
     study_file_count = study.study_files.non_primary_data.size
     share_count = study.study_shares.size
 
-    assert_equal 19, gene_count, "did not find correct number of genes"
+    # I don't need this, perhaps, because this happens a little later: assert_equal 19, gene_count, "did not find correct number of genes"
     assert_equal 1, cluster_count, "did not find correct number of clusters"
-    assert_equal 3, metadata_count, "did not find correct number of metadata objects"
+    assert_equal 22, metadata_count, "did not find correct number of metadata objects"
     assert_equal 2, cluster_annot_count, "did not find correct number of cluster annotations"
     assert_equal 3, study_file_count, "did not find correct number of study files"
     assert_equal 1, share_count, "did not find correct number of study shares"
 
+    assert_equal initial_bq_row_count + 30, get_bq_row_count(bq_dataset, study)
+
+    # request delete
+    study.study_files.each do |sf|
```

Comment:
Do we need to delete all the study files here?  Don't we just want to delete the metadata file and confirm that it cleans up?  After looking at this, this almost seems like this portion (the delete and BQ validation) should be its own  validation test that would run after this test completes and ingests data correctly.

Reply:
I'll change this. I had been thinking about checking that all the files got deleted while I was at it, but ended up not wanting to more wait time to the test.

Reply:
all set